ers/sql.go

```diff
@@ -121,11 +122,185 @@ func (rdr *SQLEventReader) Serve() (err error) {
 	return
 }
 
+// Creates mysql conditions used in WHERE statement out of filters
+func valueQry(ruleType, elem, field string, values []string, not bool) (conditions []string) {
+	// here are for the filters that their values are empty: *exists, *notexists, *empty, *notempty..
+	if len(values) == 0 {
+		switch ruleType {
+		case utils.MetaExists, utils.MetaNotExists:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s IS NOT NULL", field))
+					return
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') IS NOT NULL", elem, field))
+				return
+			}
+			if elem == utils.EmptyString {
+				conditions = append(conditions, fmt.Sprintf(" %s IS NULL", field))
+				return
+			}
+			conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') IS NULL", elem, field))
+		case utils.MetaEmpty, utils.MetaNotEmpty:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s != ''", field))
+					return
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') != ''", elem, field))
+				return
+			}
+			if elem == utils.EmptyString {
+				conditions = append(conditions, fmt.Sprintf(" %s == ''", field))
+				return
+			}
+			conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') == ''", elem, field))
+		}
+		return
+	}
+	// here are for the filters that can have more than one value: *string, *prefix, *suffix ..
+	for _, value := range values {
+		switch value { // in case we have boolean values, it should be queried over 1 or 0
+		case "true":
+			value = "1"
+		case "false":
+			value = "0"
+		}
+		var singleCond string
+		switch ruleType {
+		case utils.MetaString, utils.MetaNotString, utils.MetaEqual, utils.MetaNotEqual:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s != '%s'", field, value))
+					continue
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') != '%s'",
+					elem, field, value))
+				continue
+			}
+			if elem == utils.EmptyString {
+				singleCond = fmt.Sprintf(" %s = '%s'", field, value)
+			} else {
+				singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') = '%s'", elem, field, value)
+			}
+		case utils.MetaLessThan, utils.MetaLessOrEqual, utils.MetaGreaterThan, utils.MetaGreaterOrEqual:
+			if ruleType == utils.MetaGreaterOrEqual {
+				if elem == utils.EmptyString {
+					singleCond = fmt.Sprintf(" %s >= %s", field, value)
```

Comment:
Please do not start with empty string, put instead space when building the query.